src/sage/topology/moment_angle_complex.py

```diff
@@ -0,0 +1,781 @@
+            sage: dim(Z)
+            7
+            sage: TestSuite(Z).run()
+        """
+        # The underlying simplicial complex
+        self._simplicial_complex = copy(simplicial_complex)
+        # A dictionary of components indexed by facets
+        self._components = {}
+
+        vertices = self._simplicial_complex.vertices()
+        # it suffices to perform union only over facets
+        for facet in self._simplicial_complex.maximal_faces():
+            Y = []
+            for j in vertices:
+                if j in facet:
+                    Y.append(simplicial_complexes.Simplex(2))
+                else:
+                    Y.append(simplicial_complexes.Sphere(1))
+
+            self._components[facet] = Y
+
+    @lazy_attribute
+    def _moment_angle_complex(self):
+        """
+        Create the moment-angle complex as a cubical complex.
+
+        If this lazy attribute is accessed, we explicitly compute
+        the moment-angle complex, viewed as a cubical complex.
+
+        .. WARNING::
+
+            The construction can be very slow, it is not reccomended unless
+            the corresponding simplicial complex has 5 or less vertices.
+
+        TESTS::
+
+            sage: Z = MomentAngleComplex([[0], [1], [2]]); Z
+            Moment-angle complex of Simplicial complex with vertex set (0, 1, 2) and facets {(0,), (1,), (2,)}
+            sage: Z._moment_angle_complex
+            Cubical complex with 64 vertices and 705 cubes
+
+        This method is used in the ``cubical_complex()`` method::
+
+            sage: Z.cubical_complex()
+            Cubical complex with 64 vertices and 705 cubes
+            sage: Z.cubical_complex() == Z._moment_angle_complex
+            True
+        """
+        n = len(self._simplicial_complex.vertices())
+        D = [cubical_complexes.Cube(2)] * n
+        S = [cubical_complexes.Sphere(1)] * n
+
+        moment_angle_complex = CubicalComplex()
+        for component in self._components.values():
+            x = D[0] if component[0] == simplicial_complexes.Simplex(2) else S[0]
+            for j in range(1, len(component)):
+                y = D[j] if component[j] == simplicial_complexes.Simplex(2) else S[j]
+                x = x.product(y)
+            moment_angle_complex = union(moment_angle_complex, x)
+
+        return moment_angle_complex
+
+    def _repr_(self):
+        """
+        Return a string representation of ``self``.
+
+        TESTS::
+
+            sage: Z = MomentAngleComplex([[0,1], [1,2], [2,0]])
+            sage: Z._repr_()
+            'Moment-angle complex of Simplicial complex with vertex set (0, 1, 2) and facets {(0, 1), (0, 2), (1, 2)}'
+            sage: repr(Z)  # indiredt doctest
+            'Moment-angle complex of Simplicial complex with vertex set (0, 1, 2) and facets {(0, 1), (0, 2), (1, 2)}'
+            sage: Z  # indirect doctest
```

Comment:
```suggestion
            sage: repr(Z)
            'Moment-angle complex of Simplicial complex with vertex set (0, 1, 2) and facets {(0, 1), (0, 2), (1, 2)}'
            sage: Z
```
These are not needed.